front/app/src/Views/TheGame/TheGame.tsx

```diff
@@ -1,22 +1,165 @@
+	game.board.canvasRef = canvasRef;
+	const	update = () =>
+	{
+		game.playerOne.updatePlayerPosition();
+		game.playerTwo.updatePlayerPosition();
+		game.ball.update();
+	}
+
+	const clear = () =>
+	{
+		if (game.board.ctx)
+		{
+			game.board.ctx.fillStyle = "#fff";
+			game.board.ctx?.clearRect(0, 0, game.board.dim.width, game.board.dim.height);
+		}
+	}
+
+	const	keyHookDown = (e) =>
+	{
+		switch (e.keyCode)
+		{
+			case 38: // fleche up
+				game.actionKeyPress = 38;
+				break; 
+			case 40: // fleche down
+				game.actionKeyPress = 40;
+				break;
+			case 83: //S
+				game.actionKeyPress = 83;
+				break;
+			case 87: //W
+				game.actionKeyPress = 87;
+				break;
+			default:
+				break;
+		}
+	}
+
+	const	keyHookReleased = (e) =>
+	{
+		game.actionKeyPress = -1;
+	}
+
+	const keyEnter = (e) =>
+	{
+		game.continueAnimating = true;
+		game.startDisplayed = false;
+	}
+
+	const pauseButtonRef = useRef<HTMLInputElement>(null);
+	const resumeButtonRef = useRef<HTMLInputElement>(null);
+
 
 	useEffect(() =>
 	{
-		savePrevPage("/play-game");
+		let requestId: number;
+		const canvas = canvasRef.current;
+		const pauseButton = pauseButtonRef.current;
+		const resumeButton = resumeButtonRef.current;
+
+		const ctx = canvas?.getContext("2d");
+		game.board.canvas = canvas;
+		game.board.ctx = ctx;
+		game.board.init();
+		addEventListener("keydown", keyHookDown);
+		addEventListener("keyup", keyHookReleased);
+		addEventListener("keydown", keyEnter);
+
+		const	render = () =>
+		{
+			clear();
+			update();
+
+			game.board.ctx?.beginPath();
+			if (game.board.ctx)
+			{
+				game.board.ctx.fillStyle = "#F5F5DC";
+				game.board.ctx.fillRect(0, 0, game.board.dim.width, game.board.dim.height);
+			}
+			game.net.render();
+			game.ball.render();
+			game.playerOne.render();
+			game.playerTwo.render();
+			game.playerOne.renderScore();
+			game.playerTwo.renderScore();
+			if (game.startDisplayed == true)
+				game.displayStartMessage();
+			if (game.startDisplayed === false)
+			{
+			pauseButton?.addEventListener('click',function(){
+				game.continueAnimating = false;
+			});
+			resumeButton?.addEventListener('click',function(){
+				game.continueAnimating = true;
+			});
+			}
+			requestId = requestAnimationFrame(render);
+		};
+		requestId = requestAnimationFrame(render);
+		return (() =>
+		{
+			cancelAnimationFrame(requestId);
+		});
 	});
 
 	return (
 		<>
-			<MenuBar />
-			<p>The Game</p>
+			<div style={{textAlign: "center"}}>
+				FT_TRANSCENDANCE
+			</div>
+			<br></br>
+			<div style={{textAlign: "center"}}>
+				<canvas
+					height={game.board.canvas?.height}
+					width={game.board.canvas?.width}
+					ref={game.board.canvasRef}
+				>
+				</canvas>
+			</div>
+			<div style={{textAlign: "center"}}>
+				<input type="button" value="PAUSE" ref={pauseButtonRef} />
+				<input type="button" value="RESUME" ref={resumeButtonRef} />
+			</div>
 		</>
 	);
 };
 
-export default TheGame;
+export default GameCanvas;
```

Comment:
The exported name and className is the same, but it must match to the filename. Except case that we can use index.tsx for naming then Component will be imported from the adress of the directory